packages/babel-plugin-proposal-object-rest-spread/src/index.js

```diff
@@ -335,7 +368,7 @@ export default declare((api, opts) => {
             // skip single-property case, e.g.
             // const { ...x } = foo();
             // since the RHS will not be duplicated
-            originalPath.node.id.properties.length > 1 &&
+            doesAnyChildNodeHaveMoreThanOneProperty(originalPath.node.id) &&
             !t.isIdentifier(originalPath.node.init)
```

Comment:
Unrelated to this PR: I think we should use `originalPath.scope.isStatic(originalPath.node.init)` here, because

1) A `ThisExpression` is safe to reuse, e.g.
```js
const { ...x } = this;
```

2) Even if `node.init` is an identifier, it can not be reused if not a constant binding. e.g.

```js
let z = { v: 1 };
function sideEffect() {
  z = { v: 2 };
}
const { x = sideEffect(), ...y } = z;
y // { v: 1 }
```
`y` becomes `{ v: 2 }` after transformed because `z` can be modified in `sideEffect()`.

These two issues will be fixed by `scope.isStatic`.